packages/rest-generator/src/parser/operation.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { createRefs } from '../../test/test-util';
+import { parseOperationName, parseParameters } from './operation';
+
+describe('parseOperationName', () => {
+  it('parses the operation name from the operationId', () => {
+    expect(
+      parseOperationName({ operationId: 'testOperationId' }, 'pattern', 'get')
+    ).toEqual('testOperationId');
+  });
+
+  it('parses the operation name from the description', () => {
+    expect(
+      parseOperationName({ description: 'Do something' }, 'pattern', 'get')
+    ).toEqual('doSomething');
+  });
+
+  it('parses the operation name from the pattern and method', () => {
+    expect(parseOperationName({}, '/entity', 'get')).toEqual('getEntity');
+  });
+
+  it('parses the operation name from the pattern and method for post', () => {
+    expect(parseOperationName({}, '/entity/property', 'post')).toEqual(
+      'createEntityProperty'
+    );
+  });
+
+  it('parses the operation name from the pattern and method with one placeholders', () => {
+    expect(
+      parseOperationName({}, '/entity/{entityId}/property', 'get')
+    ).toEqual('getEntityPropertyByEntityId');
+  });
+
+  it('parses the operation name from the pattern and method with multiple placeholders', () => {
+    expect(
+      parseOperationName({}, '/entity/{entityId}/property/{propertyId}', 'get')
+    ).toEqual('getEntityPropertyByEntityIdAndPropertyId');
+  });
+
+  it('parses the operation name from the pattern and method with only placeholders', () => {
+    expect(parseOperationName({}, '/{placeholder}', 'get')).toEqual(
+      'getByPlaceholder'
+    );
+  });
+});
+
+describe('parseParameters', () => {
+  it('returns empty array if there are no parameters', async () => {
+    expect(parseParameters({}, await createRefs())).toEqual([]);
+  });
+
+  it('parses inline parameters', async () => {
+    const anyParamNoSchema = { in: 'query', name: 'anyParamNoSchema' };
+    const anyParamWithSchema = {
+      in: 'query',
+      name: 'anyParamWithSchema',
+      schema: {}
+    };
+    const stringParam = {
+      in: 'query',
+      name: 'anyParamWithSchema',
+      schema: { type: 'string' }
```

Comment:
Is string and integer the only possible parameter? Boolean? Number?

Reply:
There are others, but I thought duplicating the type mapper here would be trivial, so I just checked that for two different cases I get the expected results. Do you think that is enough?

Reply:
hm, ok we keep it this way but if we see an error related to an other type we add them :-).